core/src/wallet/tezos/transaction_builders/TezosLikeTransactionBuilder.cpp

```diff
@@ -129,15 +134,18 @@ namespace ledger {
             return ::ledger::core::TezosLikeTransactionBuilder::parseRawTransaction(currency, rawTransaction, true);
         }
 
+        // Reference: https://www.ocamlpro.com/2018/11/21/an-introduction-to-tezos-rpcs-signing-operations/
         std::shared_ptr<api::TezosLikeTransaction>
         TezosLikeTransactionBuilder::parseRawTransaction(const api::Currency &currency,
                                                          const std::vector<uint8_t> &rawTransaction,
                                                          bool isSigned) {
             auto params = currency.tezosLikeNetworkParameters.value();
             auto tx = std::make_shared<TezosLikeTransactionApi>(currency);
             BytesReader reader(rawTransaction);
-            // Watermark: Generic-Operation
-            reader.readNextByte();
+            if (!isSigned) {
+                // Watermark: Generic-Operation
+                reader.readNextByte();
```

Comment:
I think we’re going to need to make a huge pass of cleaning of our parsers because such a code can easily break libcore as we’re not ensuring that the read byte has the expected value.